### DISCUSSION TITLE
throw expression in null coalescing assignment

### DISCUSSION THREAD

**Discussion by ronnygunawan** [dotnet/csharplang#4757] [Language Ideas] [answered]
Why is this not allowed?
```cs
void F(string? s) {
    s ??= throw new ArgumentNullException(nameof(s));
}
```

Currently this code gives me an error: `CS8115: A throw expression is not allowed in this context.`

**Comment by theunrepentantgeek** [comment]
I'd conjecture it's not allowed because there's nothing assignable specified on the righthand side of the assignment. I'd further suggest this is deliberate, not an oversight.

**Comment by bernd5** [comment]
But you could write:

```cs
using System;

void F(string? s) {
    _ = s ?? throw new ArgumentNullException(nameof(s));
}
F(null);
```

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/2091.
